src/Workspaces/MSBuild/BuildHost/Build/ProjectBuildManager.cs

```diff
@@ -228,7 +235,8 @@ public void StartBatchBuild(IDictionary<string, string>? globalProperties = null
             ? [_msbuildLogger]
             : ImmutableArray<MSB.Framework.ILogger>.Empty;
 
-        _batchBuildProjectCollection = new MSB.Evaluation.ProjectCollection(allProperties, loggers, MSB.Evaluation.ToolsetDefinitionLocations.Default);
+        // https://github.com/dotnet/msbuild/issues/11867: workaround LoggerException when passing binary logger to both evaluation and build
+        _batchBuildProjectCollection = new MSB.Evaluation.ProjectCollection(allProperties, loggers: [], MSB.Evaluation.ToolsetDefinitionLocations.Default);
```

Comment:
Part of me would like to see this moved before where we create the `loggers` array. Perhaps adding a few more details to the comment will ensure no one mistakes this as a typo.

```suggestion
        // Pass empty loggers array to workaround LoggerException when passing binary logger to both evaluation and build. See https://github.com/dotnet/msbuild/issues/11867
        _batchBuildProjectCollection = new MSB.Evaluation.ProjectCollection(allProperties, loggers: [], MSB.Evaluation.ToolsetDefinitionLocations.Default);
```